Formula/pyinstaller.rb

```diff
@@ -33,6 +33,10 @@ class Pyinstaller < Formula
   end
 
   def install
+    cd "bootloader" do
+      inreplace "wscript", "mac_arch = ['-arch', 'x86_64']", "mac_arch = ['-arch', 'arm64']" if Hardware::CPU.arm?
+      system "python3", "./waf", "all", "--no-universal2", "STRIP=/usr/bin/strip"
```

Comment:
Sadly `--no-universal2` is not "use native arch" but "use x86_64".

Reply:
Can we upstream a patch?

Reply:
We may need to first discuss this with upstream as current behavior seems to be an intentional decision. May need to be designed as separate feature.
e.g.
- https://github.com/pyinstaller/pyinstaller/pull/5581
  > As this requires a recent version of Xcode toolchain (12.2 or newer), our `wafscript` now has `--no-universal2` option to enable building `x86_64`-only thin-arch bootloaders with older toolchains.
- https://github.com/pyinstaller/pyinstaller/issues/5879#issuecomment-853101735
  > Yeah, you can provide `--no-universal2` switch to `waf` and it will build regular x86_64 bootloader, as it did before # 5581 (note that even on arm64, we built x86_64 bootloader before, and we also do now if --no-universal2 switch is used.).

---

The upstream's current design seems to be providing `universal2` bootloader for most users. Then, `pyinstaller` creates frozen applications where the the underlying Python (and perhaps user options) influence whether a `universal2` application can be created, or if the `universal2` needs to be thinned out to single arch. 

The `--no-universal2` seems to have been designed for old systems rather than as a way of just creating thin bootloader.

Reply:
We could just change the `mac_arch` array in the `else` branch to
```python
mac_arch = ['-arch', platform.machine()]
```

Reply:
I opened an upstream issue https://github.com/pyinstaller/pyinstaller/issues/6091 to discuss this as the upstream comments sound like this is part of their design.

For formula work around, I'll try using a local patch instead of inreplace and see how it goes.

Reply:
Upstream discussion mention removing `-arch` flag should work, so went with a temporary patch to do this until an official solution is available.